crates/lsp/src/lsp.rs

```diff
@@ -332,11 +332,33 @@ impl LanguageServer {
             };
 
             let header = std::str::from_utf8(&buffer)?;
-            let message_len: usize = header
+            let mut segments = header.lines();
+
+            let message_len: usize = segments
+                .next()
+                .ok_or_else(|| anyhow!("unable to find the first line of the LSP message header"))? // Impossible, but just in case.
                 .strip_prefix(CONTENT_LEN_HEADER)
                 .ok_or_else(|| anyhow!("invalid LSP message header {header:?}"))?
-                .trim_end()
-                .parse()?;
+                .parse()
+                .context("failed to parse Content-Length of LSP message header")?;
+
+            if let Some(second_segment) = segments.next() {
```

Comment:
Curious to know the reason for changes here: was it the first ever server to actually send content-type to us?

Reply:
Yep, unfortunately was! see #7220 for another LSP that does it as well.

Reply:
Oh, thank you for biting the bullet for us and fixing it.

Reply:
I'm not handling the actual value of the Content-Type however, is it fine? 

Seemed useless as they wouldn't have gotten past certain points if they were anything but application/vscode-jsonrpc and utf(-)8.

Reply:
Given that we did not handle anything before that and were fine — sure, you can leave it as is.
Some follow-up PRs on the topic are welcome, but seems not necessary yet too.